winlogbeat/module/pipeline.go

```diff
@@ -0,0 +1,211 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package module
+
+import (
+	"embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/joeshaw/multierror"
+	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/beats/v7/filebeat/fileset"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/esleg/eslegclient"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+var PipelinesFS *embed.FS
+var errNoFS = errors.New("no ebedded file system")
+
+const logName = "pipeline"
+
+type pipeline struct {
+	id       string
+	contents map[string]interface{}
+}
+
+// Upload reads all pipelines stored in winlogbeat executable adapts
+// pipeline for a given ES version, converts to JSON if necessary and
+// creates or updates ingest pipeline in ES
+func UploadPipelines(info beat.Info, esClient *eslegclient.Connection, overwritePipelines bool) error {
+	log := logp.NewLogger(logName)
+	pipelines, err := readAll(info)
+	if err != nil {
+		return err
+	}
+	return load(esClient, pipelines, overwritePipelines, log)
+}
+
+// Export reads all pipelines stored in winlogbeat executable adapts
+// pipelines for a given ES version and writes the converted pipelines
+// to the given directory in JSON format
+func ExportPipelines(info beat.Info, version common.Version, directory string) error {
+	if !info.ElasticLicensed {
```

Comment:
Can we remove this check?. I never like to add this into OSS code if it can be avoided. It looks like if just this is removed then it will return nil since `readAll` will return nil.

Reply:
Yep, removing.  It was an oversight that I left it in.